### PR TITLE
Fix for engineering borgs reclaiming their metal sheets

### LINKED ISSUE #2463: Engineering borgs Reclaim their metal
[BUG]

**Describe the bug**
Engineering borgs can put their metal sheets and glass sheets and probably wires into the portable reclaimer and thus removing it all
**To Reproduce**
1. be engi borg
2. click reclaimer with metal active
3. metal goes into thing, activate reclaimer, you no longer have metal sheets

**Expected behavior**
not put sheets in reclaimer

### PR DESCRIPTION
**Pull request by j-awn**
[BUG]
## About the PR 

Cyborgs used to be able to stuff their metal sheets into the portable reclaimer. Now they can't; the reclaimer checks to see if the item can be dropped before it takes it.

## Why's this needed? 

Fixes #2463, and for good measure the same fix is applied to the Material processor (`/obj/machinery/processor`)

## Changelog

N/A

### CODE REVIEW

File: code/obj/item/material.dm [hunk 1]
```diff
@@ -841,7 +841,10 @@
 		playsound(src.loc, sound_process, 40, 1)
 
 	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W,/obj/item/raw_material/) || istype(W,/obj/item/sheet/) || istype(W,/obj/item/rods/) || istype(W,/obj/item/tile/) || istype(W,/obj/item/cable_coil) || istype(W,/obj/item/wizard_crystal))
+		if (W.cant_drop) //For borg held items
+			user.show_text("You can't put that in [src] when it's attached to you!", "red")
+			return ..()
+		else if (istype(W,/obj/item/raw_material/) || istype(W,/obj/item/sheet/) || istype(W,/obj/item/rods/) || istype(W,/obj/item/tile/) || istype(W,/obj/item/cable_coil) || istype(W,/obj/item/wizard_crystal))
```
Comment (mordent-goonstation) [inline code/obj/item/material.dm:847]: No need to `else` if you're returning in the previous `if` above.

```suggestion
		if (istype(W,/obj/item/raw_material/) || istype(W,/obj/item/sheet/) || istype(W,/obj/item/rods/) || istype(W,/obj/item/tile/) || istype(W,/obj/item/cable_coil) || istype(W,/obj/item/wizard_crystal))
```

File: code/modules/materials/Mat_Processing.dm [hunk 2]
```diff
@@ -87,6 +87,10 @@
 			S.satchel_updateicon()
 			return
 
+		if (W.cant_drop) //For borg held items
+			user.show_text("You can't put that in [src] when it's attached to you!", "red")
```
Comment (mordent-goonstation) [inline code/modules/materials/Mat_Processing.dm:91]: Prefer `boutput` to `show_text`, I think? (See example a few lines down)

I'm unsure of the differences at this time, but I see `boutput` a _lot_ more.

File: code/obj/item/material.dm [hunk 3]
```diff
@@ -841,7 +841,10 @@
 		playsound(src.loc, sound_process, 40, 1)
 
 	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W,/obj/item/raw_material/) || istype(W,/obj/item/sheet/) || istype(W,/obj/item/rods/) || istype(W,/obj/item/tile/) || istype(W,/obj/item/cable_coil) || istype(W,/obj/item/wizard_crystal))
+		if (W.cant_drop) //For borg held items
+			user.show_text("You can't put that in [src] when it's attached to you!", "red")
```
Comment (mordent-goonstation) [inline code/obj/item/material.dm:845]: `boutput`

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

File: code/modules/materials/Mat_Processing.dm (hunk 2 above)
Comment (j-awn) [reply code/modules/materials/Mat_Processing.dm:91]: this is cargo cult'd from a similar fix I did to kitchen stuff a few weeks ago. if you feel strongly though I can change it

File: code/obj/machinery/deep_fryer.dm [hunk 4]
```diff
@@ -26,7 +26,7 @@
 			boutput(usr, "<span class='alert'>The [src] refuses to interface with you, as you are not a properly trained chef!</span>")
 			return
 		if (W.cant_drop) //For borg held items
-			user.show_text("You can't put that in [src] when it's attached to you!", "red")
+			boutput(user, "You can't put that in [src] when it's attached to you!")
```
Comment (mordent-goonstation) [inline code/obj/machinery/deep_fryer.dm:29]: This (and the other kitchen stuff ones) aren't really related to this PR. The code you've replaced is also not comparable, as the previous stuff is "red" (so you'd want something like `boutput(user, "<span class='alert'>You can't put that in [src] when it's attached to you!</span>"). (See three lines below this one)

Personal take: split into a second PR. Lots of little, unrelated PRs is better than one PR with multiple unrelated changes.

File: code/obj/item/material.dm [hunk 5]
```diff
@@ -841,6 +841,9 @@
 		playsound(src.loc, sound_process, 40, 1)
 
 	attackby(obj/item/W as obj, mob/user as mob)
+		if (W.cant_drop) //For borg held items
+			boutput(user, "You can't put that in [src] when it's attached to you!")
```
Comment (mordent-goonstation) [inline code/obj/item/material.dm:845]:
```suggestion
			boutput(user, "<span class='alert'>You can't put that in [src] when it's attached to you!</span>")
```

File: code/modules/materials/Mat_Processing.dm [hunk 6]
```diff
@@ -87,6 +87,10 @@
 			S.satchel_updateicon()
 			return
 
+		if (W.cant_drop) //For borg held items
+			boutput(user, "You can't put that in [src] when it's attached to you!")
```
Comment (mordent-goonstation) [inline code/modules/materials/Mat_Processing.dm:91]:
```suggestion
			boutput(user, "<span class='alert'>You can't put that in [src] when it's attached to you!</span>")
```

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]

**Review by ZeWaka — APPROVED** [review]

**Review by mordent-goonstation — APPROVED** [review]

**Pull request merged**